static/src/js/containers/SavedProjectsContainer/__tests__/SavedProjectsContainer.test.jsx

```diff
@@ -0,0 +1,228 @@
+      authToken: 'mock-token'
+    }
+
+    expect(mapStateToProps(store)).toEqual({
+      authToken: 'mock-token',
+      earthdataEnvironment: 'prod-env'
+    })
+  })
+})
+
+describe('SavedProjectsContainer', () => {
+  describe('When the component mounts with a valid authToken', () => {
+    test('fetches the projects and displays them', async () => {
+      const mockProjects = [
+        {
+          id: '123',
+          name: 'Project 1',
+          path: '/search?p=!C123',
+          created_at: '2022-01-01T00:00:00.000Z'
+        }
+      ]
+
+      nock(/localhost/)
+        .get(/projects/)
+        .reply(200, mockProjects)
+
+      setup()
+
+      expect(await screen.findByText('Project 1')).toBeInTheDocument()
+      expect(screen.getByRole('button', { name: /remove project/i })).toBeInTheDocument()
+    })
+  })
+
+  describe('When the authToken is null', () => {
+    test('renders no projects', async () => {
+      setup({
+        overrideProps: {
+          authToken: null
+        }
+      })
+
+      await waitFor(() => {
+        expect(screen.queryByRole('button', { name: /remove project/i })).toBeNull()
+      })
+    })
+  })
+
+  describe('When deleting a project', () => {
+    const projectId = '123'
+
+    test('successfully deletes the project and shows a success toast', async () => {
+      const mockProjects = [
+        {
+          id: projectId,
+          name: 'Project 1',
+          path: '/search?p=!C123',
+          created_at: '2022-01-01T00:00:00.000Z'
+        }
+      ]
+
+      nock(/localhost/)
+        .get(/projects/)
+        .reply(200, mockProjects)
+
+      nock(/localhost/)
+        .delete(`/projects/${projectId}`)
+        .reply(204)
+
+      window.confirm = jest.fn(() => true)
+
+      setup()
+
+      const removeBtn = await screen.findByRole('button', { name: /remove project/i })
+
+      // eslint-disable-next-line testing-library/no-unnecessary-act
+      await act(async () => {
+        await userEvent.click(removeBtn)
+      })
+
+      expect(addToast).toHaveBeenCalledTimes(1)
+      await waitFor(() => {
+        expect(addToast).toHaveBeenCalledWith('Project removed', {
+          appearance: 'success',
+          autoDismiss: true
+        })
+      })
+    })
+
+    test('handles an error during project deletion', async () => {
+      const mockProjects = [
+        {
+          id: projectId,
+          name: 'Project 1',
+          path: '/search?p=!C123',
+          created_at: '2022-01-01T00:00:00.000Z'
+        }
+      ]
+
+      nock(/localhost/)
+        .get(/projects/)
+        .reply(200, mockProjects)
+
+      nock(/localhost/)
+        .delete(`/projects/${projectId}`)
+        .replyWithError('Failed to delete')
+
+      window.confirm = jest.fn(() => true)
+
+      const { props } = setup()
+
+      const removeBtn = await screen.findByRole('button', { name: /remove project/i })
+      await userEvent.click(removeBtn)
+
+      await waitFor(() => {
```

Comment:
You don't need to wrap both assertions in `waitFor`, I took both of the `waitFor` in this test out with no issues.

Reply:
Why was this change reverted?